arrow/constants.py

```diff
@@ -1,9 +1,9 @@
-# -*- coding: utf-8 -*-
-
 # Output of time.mktime(datetime.max.timetuple()) on macOS
 # This value must be hardcoded for compatibility with Windows
 # Platform-independent max timestamps are hard to form
 # https://stackoverflow.com/q/46133223
-MAX_TIMESTAMP = 253402318799.0
+from datetime import datetime
+
+MAX_TIMESTAMP = datetime.max.timestamp()
```

Comment:
Are we sure this is the right change here, won't this result in the bug mentioned in the comment above?

Reply:
Yeah, reverted the change for now.

Maybe we can do this with a try/catch or with os detection? so Windows can have hardcoded and the rest use something sane.

Reply:
implemented with os.name. :+1: